### DISCUSSION TITLE
[Art competition] Abstract Interpretation

### DISCUSSION THREAD

**Discussion by forceow** [prosyslab-classroom/cs524-program-analysis#280] [Art competition]
![DALL·E 2022-11-29 13 59 07 - an oil painting of the pawn dreaming of being the queen and the knight in chess](https://user-images.githubusercontent.com/91929599/204776543-74abc0a1-d0e6-43e7-8041-108883e7bf60.png)

## Short Description
For me, [Abstract Interpretation] means [an oil painting of the pawn dreaming of being the queen and the knight in chess]".

## Long Description (up to 3 sentences)
In a game of thess, a pawn can promote to whatever it wants (except for the king, of course) based on the strategy, to enable stronger movements. Similar process goes for abstract interpretation, as the technique effectively translates given values to the selected domain based on a proper choice, and allows us to perform robust analysis. "Promoting" values to proper domains would be a dream of every researchers, and even for values themselves, who knows!

**Comment by KihongHeo** [comment]
By "promoting", do you mean "abstraction"?

**Comment by forceow** [comment]
Exactly!

**Comment by KihongHeo** [comment]
Interesting. Why do you think abstraction is a promotion? Does it actually lose something, no? (e.g., precision)

**Comment by forceow** [comment]
When a pawn promotes, though it loses its trait of going straightforward, it is granted a power to move freely than before. For the case of abstract interpretation, similar holds, because with some trade-offs we can be in a better situation for analysis.
